webpack/assets/javascripts/react_app/redux/reducers/notifications/index.js

```diff
@@ -30,16 +29,18 @@ const hasUnreadMessages = notifications => {
   return result;
 };
 
-export default (state = initialState, action) => {
-  const { payload } = action;
+const { SUCCESS, FAILURE } = actionTypeGenerator(NOTIFICATIONS);
 
-  switch (action.type) {
-    case NOTIFICATIONS_POLLING_STARTED:
-      return state.set('isPolling', true);
-    case NOTIFICATIONS_GET_NOTIFICATIONS:
-      return state
-        .set('notifications', payload.notifications)
-        .set('hasUnreadMessages', hasUnreadMessages(payload.notifications));
+export default (state = initialState, { type, payload, response }) => {
+  switch (type) {
+    case SUCCESS:
+      return state.merge({
+        notifications: response.notifications,
+        hasUnreadMessages: hasUnreadMessages(response.notifications),
+      });
+    case FAILURE: {
+      return state.set('error', response);
+    }
```

Comment:
does it make sense to use API reducer here?

Reply:
it is a bit tricky, since the notifications are consumed also on other cases in this reducer,
and the provided state doesn't give access to the API reducer, a workaround I can do is to use `getState()` from the action and use a selector to retrieve the notifications data.

what do you think about that?

Reply:
another option is to move the logic to selectors

Reply:
third option is to wait for all types of API operations to be supported - #7261 
and replace all of the notifications actions with API calls.

the API calls will update the API reducer anyway and only than we won't need those cases in the reducer. imo that would be the best option.

Reply:
:+1: to the third option